runtime/fundamentals/linting_and_formatting.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Linting and Formatting"
+---
+
+In an ideal world, your code is always clean, consistent, and free of pesky
+errors. That’s the promise of Deno’s built-in linting and formatting tools. By
+integrating these features directly into the runtime, Deno eliminates the need
+for external dependencies and complex configurations in your projects. These
+inbuilt tools are fast and performant, only saving time but also ensuring that
+every line of code adheres to best practices.
+
+With `deno fmt` and `deno lint`, you can focus on writing great code, knowing
+that Deno has your back. It’s like having a vigilant assistant who keeps your
+codebase in top shape, allowing you to concentrate on what truly matters:
+building amazing applications.
+
+## Linting
+
+Linting is the process of analyzing your code for potential errors, bugs, and
+stylistic issues. Deno’s built-in linter, [`deno lint`](TODO:lint-cli-ref),
+supports recommended set of rules from [ESLint](https://eslint.org/) to provide
+comprehensive feedback on your code. This includes identifying syntax errors,
+enforcing coding conventions, and highlighting potential issues that could lead
+to bugs.
+
+To run the linter, use the following command in your terminal:
+
+```bash
+deno lint
+```
+
+By default, `deno lint` analyzes all TypeScript and JavaScript files in the
+current directory and its subdirectories. If you want to lint specific files or
+directories, you can pass them as arguments to the command. For example:
+
+```bash
+deno lint src/
+```
+
+This command will lint all files in the `src/` directory.
+
+The linter can be configured in a [`deno.json`](TODO:config-link) file. You can
+specify custom rules, plugins, and settings to tailor the linting process to
+your needs.
+
+## Formatting
+
+Formatting is the process of automatically adjusting the layout of your code to
+adhere to a consistent style. Deno’s built-in formatter, `deno fmt`, uses the
+powerful [Prettier](https://prettier.io/) engine to ensure that your code is
+always clean, readable, and consistent.
+
+To format your code, simply execute the following command in your terminal:
+
+```bash
+deno fmt
+```
+
+By default, `deno fmt` formats all TypeScript and JavaScript files in the
+current directory and its subdirectories. If you want to format specific files
+or directories, you can pass them as arguments to the command. For example:
+
+```bash
+deno fmt src/
+```
+
+This command will format all files in the `src/` directory.
```

Comment:
I wonder if we should add a note about `deno fmt --check` to check formatting on CI. That's likely the thing people want to know about to ensure consistent formatting. Or maybe we should have something in this doc that points to how to set up Deno on CI.

Reply:
I've added in an extra section